server/base/src/main/java/org/apache/accumulo/server/fs/VolumeManagerImpl.java

```diff
@@ -387,17 +387,25 @@ public short getDefaultReplication(Path path) {
   private static Configuration getVolumeManagerConfiguration(AccumuloConfiguration conf,
       final Configuration hadoopConf, final String filesystemURI) {
 
-    final Configuration volumeConfig = new Configuration(hadoopConf);
-
-    conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX).entrySet()
-        .stream().filter(e -> e.getKey().startsWith(filesystemURI + ".")).forEach(e -> {
-          String key = e.getKey().substring(filesystemURI.length() + 1);
-          String value = e.getValue();
-          log.info("Overriding property {} for volume {}", key, value, filesystemURI);
-          volumeConfig.set(key, value);
-        });
-
-    return volumeConfig;
+    Map<String,String> volumeHdfsConfigOverrides =
+        conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX);
+
+    // Calling new Configuration(Configuration) will synchronize on the constructor parameter
+    // and can cause issues if many threads call this method concurrently.
+    if (!volumeHdfsConfigOverrides.isEmpty()) {
+      final Configuration volumeConfig = new Configuration(hadoopConf);
+
+      volumeHdfsConfigOverrides.entrySet().stream()
+          .filter(e -> e.getKey().startsWith(filesystemURI + ".")).forEach(e -> {
```

Comment:
Would it be worthwhile to avoid creating the hadoop config copy in the case where filter on the stream drops everything?

Reply:
Yeah, so I'm trying to add that right now and running into compilation issues. It's turning the resulting map into Map<Object,Object> vs Map<String,String>. Do you have something that works?

Reply:
```
    Map<String,String> volumeHdfsConfigOverrides = 
        conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX).entrySet()
            .stream().filter(e -> e.getKey().startsWith(filesystemURI + "."))
            .collect(Collectors.toUnmodifiableMap(Function.identity(), Function.identity()));
```

doesn't compile.

Reply:
I got it:

```
    Map<String,String> volumeHdfsConfigOverrides = 
        conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX).entrySet()
            .stream().filter(e -> e.getKey().startsWith(filesystemURI + "."))
            .collect(Collectors.toUnmodifiableMap(e -> e.getKey(), e -> e.getValue()));
```

Reply:
Could do something like

```java
    List<Entry<String, String>> matching = conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX).entrySet()
            .stream().filter(e -> e.getKey().startsWith(filesystemURI + ".")).collect(Collectors.toList());
    
    if(!matching.isEmpty()) {
      Configuration volumeConfig = new Configuration(hadoopConf);
      matching.forEach(e->{
        String key = e.getKey().substring(filesystemURI.length() + 1);
        String value = e.getValue();
        log.info("Overriding property {} for volume {}", key, value, filesystemURI);
        volumeConfig.set(key, value);
      });
    }
```

Reply:
This works: 

> I got it:
> 
> ```
>     Map<String,String> volumeHdfsConfigOverrides = 
>         conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX).entrySet()
>             .stream().filter(e -> e.getKey().startsWith(filesystemURI + "."))
>             .collect(Collectors.toUnmodifiableMap(e -> e.getKey(), e -> e.getValue()));
> ```

I was just about to post it but a little different:

```
    Map<String,String> volumeHdfsConfigOverrides2 =
            conf.getAllPropertiesWithPrefixStripped(Property.INSTANCE_VOLUME_CONFIG_PREFIX).entrySet()
                    .stream().filter(e -> e.getKey().startsWith(filesystemURI + "."))
                    .collect(Collectors.toUnmodifiableMap(Entry::getKey, Entry::getValue));
```